src/box/lua/upgrade.lua

```diff
@@ -314,9 +315,34 @@ space_formats_1_7_5._vuser = space_formats_1_7_5._user
 space_formats_1_7_5._vpriv = space_formats_1_7_5._priv
 
 local function space_trig_1_7_5(_, tuple)
-    if tuple and space_formats_1_7_5[tuple[3]] and
+    if tuple == nil then
+        return nil
+    end
+    local is_changed = false
+    if space_formats_1_7_5[tuple[3]] and
        not table.equals(space_formats_1_7_5[tuple[3]], tuple[7]) then
+        -- Update format of system spaces.
         tuple = tuple:update{{'=', 7, space_formats_1_7_5[tuple[3]]}}
+        is_changed = true
+    elseif tuple[7] and type(tuple[7]) == 'table' and next(tuple[7]) ~= nil and
```

Comment:
Ditto, `type(tuple[7]) == 'table'` already filters out `nils`, so the previous check of `elseif tuple[7]` isn't needed.

Reply:
Fixed